### PR TITLE
Added short-hand Event->call() for easier event calling

### PR DESCRIPTION
**Pull request by dktapps**
## Introduction
`$this->server->getPluginManager()->callEvent($ev)` is long and annoying. `$ev->call()` is a much nicer and quicker short-form that can now be used as an alternative.

This PR is still a work in progress since there are a very large number of these event calls scattered throughout the code. A sample has been supplied as an example. If everyone is happy with it I will convert the remaining usages.

### API changes
- Added `Event->call()` which can be substituted for long `blah->getBlah()->getBlah()->getServer()->getPluginManager()->...` calls.

## Backwards compatibility
No BC breaks, only API additions.

### CODE REVIEW

**Comment by TheDiamondYT1** [comment]
Maybe deprecate `PluginManager->callEvent()`?

**Comment by dktapps** [comment]
No, this is simply syntactic sugar for developers.

**Comment by TheDiamondYT1** [comment]
@dktapps Okay

File: src/pocketmine/Player.php [hunk 1]
```diff
@@ -1086,7 +1086,7 @@ public function dataPacket(DataPacket $packet, $needACK = false){
 		$timings = Timings::getSendDataPacketTimings($packet);
 		$timings->startTiming();
 
-		$this->server->getPluginManager()->callEvent($ev = new DataPacketSendEvent($this, $packet));
+		($ev = new DataPacketSendEvent($this, $packet))->call();
 		if($ev->isCancelled()){
```
Comment (SOF3) [inline src/pocketmine/Player.php]: What about also returning isCancelled() in the return value, or a constant `false` if the event is not a Cancellable?

**Review by SOF3 — CHANGES_REQUESTED** [review]

**Review by SOF3 — APPROVED** [review]

**Comment by TheDiamondYT1** [comment]
When will this be merged?

File: src/pocketmine/Player.php (hunk 1 above)
Comment (SOF3) [inline src/pocketmine/Player.php]: Then what about returning `$this`?

**Comment by dktapps** [comment]
This syntax, while shorter, is more ugly than the current approach. This branch also has conflicts with master that I have no interest in resolving, so I'm going to close it.

**Comment by jasonwynn10** [comment]
:frowning:

**Pull request closed without merging**